### PR TITLE
Add Redis subchart

### LINKED ISSUE #79: Upgrade chart testing action
https://github.com/helm/chart-testing-action/releases/tag/v1.0.0-rc.1

Brings in helm 3 support.

### LINKED ISSUE #140: Add redis chart dependency and settings
discussed with @bjoernw 

Rough requirements:

- Work correctly in insecure mode
- Use TLS under normal circumstances
- Possibly generate TLS certs automatically
- Use banzai redis upstream non-cluster, non-sentinal mode (we don't support that yet)
- Only install redis chart if requested by the user explicitly

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary
Add a basic redis subchart.

- Supports automatic TLS out of the box
- Supports insecure mode for service mesh deployment
- Configures basic redis instance sufficient for low volume / development environments

Also, update chart testing to `v1.0.0` to support helm 3 tests.

## Related issues
Closes #140 
Closes #79

**Checklist**:
- [x] add related issues
- [x] update Configuration in README
- [x] ready for review

### CODE REVIEW

File: charts/pomerium/values.yaml [hunk 1]
```diff
@@ -284,3 +284,18 @@ serviceMonitor:
 
 rbac:
   create: true
+
+redis:
+  enabled: false
+  existingSecret: pomerium-redis-password
```
Comment (travisgroth) [inline charts/pomerium/values.yaml:290]: Subchart values cannot be templated to use the release name with `_helpers.tpl` due to helm limitations.

File: charts/pomerium/values.yaml [hunk 2]
```diff
@@ -284,3 +284,18 @@ serviceMonitor:
...
+  existingSecretPasswordKey: password
+  generateTLS: true
+  forceGenerateTLS: false
+  cluster:
+    slaveCount: 1
+  tls:
+    enabled: true
+    certificatesSecret: pomerium-redis-tls
```
Comment (travisgroth) [inline charts/pomerium/values.yaml:298]: Subchart values cannot be templated to use the release name with `_helpers.tpl` due to helm limitations.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -23,20 +32,20 @@ jobs:
...
 
       - name: Run chart-testing (install)
-        uses: helm/chart-testing-action@v1.0.0-alpha.3
+        uses: helm/chart-testing-action@v1.0.0
         with:
           command: install
           config: .ct.yaml
+
+      - name: Rebuild deps with helm 2
+        run: docker run -i -v $(pwd):/work quay.io/helmpack/chart-testing:v2.4.0 /bin/sh -c 'cd /work/charts/pomerium/ && helm init --client-only && helm repo add bitnami https://charts.bitnami.com/bitnami && helm dep update && helm dep build'
```
Comment (travisgroth) [inline .github/workflows/test.yaml:45]: This is gross but it'll rebuild the dependency metadata for helm v2 for later testing.

**Review by desimone — DISMISSED** [review]
I haven't had a chance to test on my environment yet but LGTM. 

Added a few nits, take or leave them.

File: charts/pomerium/Chart.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: pomerium
-version: 12.2.4
+version: 12.3.0
 appVersion: 0.10.3
```
Comment (desimone) [inline charts/pomerium/Chart.yaml:4]: Bump!

**Review by desimone — DISMISSED** [review]
Tested and confirmed working.

File: charts/pomerium/Chart.yaml (hunk 4 above)
Comment (travisgroth) [reply charts/pomerium/Chart.yaml:4]: Will be fixed in another PR.

**Review by desimone — APPROVED** [review]

**Pull request merged**